### PR TITLE
Feature/monitoring smr

### PR DESCRIPTION
**Pull request by Anny777**

### CODE REVIEW

File: tests/Tenders.API.Tests/EmailSmrMonitoringStrategyTest.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        private readonly Mock<IEmailFactory<SmrMailingProcess>> emailFactory = new Mock<IEmailFactory<SmrMailingProcess>>();
+        private readonly Mock<IMailProcessRepository> repository = new Mock<IMailProcessRepository>();
+        private readonly IStrategy<IEmail> sut;
+
+        private readonly IEmail smrEmail = new SmrMailingProcess
+        {
+            Subject = "Test",
+            Body = "Test body",
+            Receivers = new[] { "Reciever_1@mail.ru", "Reciever_2@mail.ru" },
```
Comment (DamienVStone) [inline tests/Tenders.API.Tests/EmailSmrMonitoringStrategyTest.cs]: contoso.com

File: tests/Tenders.API.Tests/MailingProcessStrategyTest.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        private readonly Mock<IEmailFactory<MailingProcess>> emailFactory = new Mock<IEmailFactory<MailingProcess>>();
+        private readonly Mock<IMailProcessRepository> repository = new Mock<IMailProcessRepository>();
+        private readonly IStrategy<IEmail> sut;
+
+        private readonly IEmail smrEmail = new MailingProcess
+        {
+            Subject = "Test",
+            Body = "Test body",
+            Receivers = new[] { "Reciever_1@mail.ru", "Reciever_2@mail.ru" },
```
Comment (DamienVStone) [inline tests/Tenders.API.Tests/MailingProcessStrategyTest.cs]: contoso.com

File: tests/Tenders.Synchronization.SmrMonitoring.Tests/ServiceRegistrationPointTest.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    /// <summary>
+    /// Testing service registration point.
+    /// </summary>
+    public class ServiceRegistrationPointTest
+    {
+        /// <summary>
+        /// Checks the resolve.
+        /// </summary>
+        [Fact]
```
Comment (DamienVStone) [inline tests/Tenders.Synchronization.SmrMonitoring.Tests/ServiceRegistrationPointTest.cs]: remove fact

File: tests/Tenders.Synchronization.SmrMonitoring.Tests/SmrMonitoringServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+               .Setup(c => c.MakeExtendedSearchUri(It.IsAny<string>()))
+               .Returns(new Uri("https://contoso.com/search"));
+
+            this.smrConfig
+                .Setup(c => c.MakeGetInfoUri(It.IsAny<string>()))
+                .Returns(new Uri("https://contoso.com/info"));
+
+            this.participant = new Participant()
+            {
+                Name = "КАЗЁННЫЙ ТЕСТ",
```
Comment (DamienVStone) [inline tests/Tenders.Synchronization.SmrMonitoring.Tests/SmrMonitoringServiceTest.cs]: !

File: API/Tenders.API.DAL.Postgres/ApplicationContext.cs [hunk 5]
```diff
@@ -141,6 +140,7 @@ protected override void OnModelCreating(ModelBuilder builder)
             builder.Entity<TenderPlan223>();
 
             builder.Entity<MailingProcess>();
+            builder.Entity<SmrMailingProcess>();
```
[the remark this answers is not in the document]
Comment (Anny777) [reply API/Tenders.API.DAL.Postgres/ApplicationContext.cs]: It is not supported in EF 6.

File: Synchronization/Tenders.Synchronization.SmrMonitoring/Services/SmrMonitoringService.cs [hunk 6]
```diff
@@ -67,13 +67,20 @@ internal class SmrMonitoringService : IScanner
...
-            await this.GetGeneralInfo(ct);
+            try
+            {
+                ct.ThrowIfCancellationRequested();
+                await this.GetGeneralInfo(ct);
+            }
+            catch (Exception ex)
+            {
+                await this.loggerService.Error(ex.InnerException.Message);
+            }
```
Comment (DamienVStone) [inline Synchronization/Tenders.Synchronization.SmrMonitoring/Services/SmrMonitoringService.cs]: Add throw

**Pull request merged**